### PR TITLE
feat: support as a pure api service

### PR DESCRIPTION
**Pull request by yimingjfe**
# PR Details

Modernjs 工程项目支持纯粹的 API 服务

## Description

主要修改点：

1、当只有 api 目录时，不执行前端的一些逻辑；如 webpack 构建，前端路由注入等。
2、将 route.json 的生成逻辑，从 webpack plugin 改为放到一个函数中；不依赖 webpack。
3、修改编译完成或启动 server 后的展示逻辑，支持纯 api 服务时；告知用户服务已启动完成。

## Related Issue

https://github.com/modern-js-dev/modern.js/issues/519
https://github.com/modern-js-dev/modern.js/issues/524

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by leeight — CHANGES_REQUESTED** [review]
这个改动最好补充 Integration Test 的测试用例

File: packages/cli/plugin-analyze/src/index.ts [hunk 1]
```diff
@@ -76,56 +76,72 @@ export default createPlugin(
...
+          routes: [],
+        });
+
+        debug(`server routes: %o`, routes);
+
+        AppContext.set({
+          ...appContext,
+          existSrc,
+          serverRoutes: routes,
+        });
```
Comment (leeight) [inline packages/cli/plugin-analyze/src/index.ts:143]: 其实可以改成

```
if (!existSrc) {

        const { routes } = await (mountHook() as any).modifyServerRoutes({
          routes: [],
        });

        debug(`server routes: %o`, routes);

        AppContext.set({
          ...appContext,
          existSrc,
          serverRoutes: routes,
        });
  return;
}
```

避免这么多的 diff 产生

File: packages/solutions/app-tools/tests/commands/build.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  __esModule: true,
+  mountHook() {
+    return {
+      beforeBuild: mockBeforeBuild,
+      afterBuild: mockAfterBuild,
+    };
+  },
+  useAppContext: jest.fn(() => ({
+    appDirectory: {
+      existSrc: false,
```
Comment (zllkjc) [inline packages/solutions/app-tools/tests/commands/build.test.ts:17]: 这里啥意思呀，为啥会在 appDirectory 里

Comment (yimingjfe) [reply packages/solutions/app-tools/tests/commands/build.test.ts]: 这里写错了，已修改。

**Review by zllkjc — APPROVED** [review]

**Comment by yimingjfe** [comment]
> 这个改动最好补充 Integration Test 的测试用例

已补充在 tests/integration/api-service-koa 下，非 API 服务项目由之前的集成用例可以保证。

**Pull request merged**